gokitmiddlewares/loggingmiddleware/loggercontext.go

```diff
@@ -0,0 +1,61 @@
+package loggingmiddleware
+
+import (
+	"context"
+	"time"
+
+	"github.com/arquivei/foundationkit/request"
+	"github.com/arquivei/foundationkit/trace"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
+
+func initLoggerContext(ctx context.Context, l zerolog.Logger) context.Context {
+	// Creates a copy, otherwise the logger is updated for every request
+	// Ensures that there is a logger in the context
+	// If there is a logger in the context, it's not updated
```

Comment:
The zerolog documentation is ambiguous, but actually the context is not updated only if `l` is Disabled or if `l` is the same logger as the one already registered. If it's another logger instance, it will return a new context with a the `l` received

```
func (l *Logger) WithContext(ctx context.Context) context.Context {
	if lp, ok := ctx.Value(ctxKey{}).(*Logger); ok {
		if lp == l {
			// Do not store same logger.
			return ctx
		}
	} else if l.level == Disabled {
		// Do not store disabled logger.
		return ctx
	}
	return context.WithValue(ctx, ctxKey{}, l)
}
```